src/core/lightning/index.ts

```diff
@@ -305,14 +311,20 @@ export const LightningMixin = (superclass) =>
               await cancelHodlInvoice({ lnd, id })
               this.logger.info({ id, user: this.user }, "canceling invoice on lnd")
 
-              const resultUpdate = await InvoiceUser.updateOne(
-                { _id: id },
-                { paid: true },
-              )
-              this.logger.info(
-                { id, user: this.user, resultUpdate },
-                "invoice has been updated from InvoiceUser following on_us transaction",
-              )
+              const isUpdated = await this.invoices.setPaidByPaymentHash(id)
```

Comment:
This is not the right way to use a repository.
It should be:
```
wallet = invoices.findByPaymentHash(hash)
wallet.paid = true
invoices.persist(wallet)
```

What you are doing here is putting use case code into the repository. But repository should only know about persistence details. What kind of data transformation is required to set a wallet as paid is business logic concern that belongs in domain / or if its trivial can be executed directly on domain objects.

Reply:
we already have some performance issues. how do you suggest to do it without affect more the performance?

Reply:
What are the performance issues?